packages/odyssey-react-mui/src/labs/Switch.tsx

```diff
@@ -0,0 +1,238 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  memo,
+  SyntheticEvent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import {
+  Switch as MuiSwitch,
+  CheckboxProps as MuiCheckboxProps,
+  FormControlLabel,
+} from "@mui/material";
+
+import { useOdysseyDesignTokens } from "../OdysseyDesignTokensContext";
+import { Box } from "../Box";
+import { FieldComponentProps } from "../FieldComponentProps";
+import { FieldHint } from "../FieldHint";
+import type { SeleniumProps } from "../SeleniumProps";
+import { useUniqueId } from "../useUniqueId";
+import { ComponentControlledState, getControlState } from "../inputUtils";
+import { CheckedFieldProps } from "../FormCheckedProps";
+
+type OnChangeCallbackArguments = {
+  checked: boolean;
+  value: string;
+};
+
+export type SwitchProps = {
+  /**
+   * if `true`, the label and switch span entire containing element's width
+   */
+  isFullWidth?: boolean;
+  /**
+   * The label text for the Switch
+   */
+  label: string;
+  /**
+   * The change event handler for the Switch
+   */
+  onChange?: ({ checked, value }: OnChangeCallbackArguments) => void;
+  /**
+   * The value attribute of the Switch
+   */
+  value: string;
+} & Pick<
+  FieldComponentProps,
+  "hint" | "id" | "isFullWidth" | "isDisabled" | "name"
+> &
+  CheckedFieldProps<MuiCheckboxProps> &
```

Comment:
Just checking... does this pull in the `checked` and `isDefaultChecked` props?

Reply:
This pulls in `isChecked` and `isDefaultChecked`